persistence/provider/memory/eventstore.go

```diff
@@ -42,14 +41,52 @@ func (r *eventStoreRepository) NextEventOffset(
 	return uint64(next), nil
 }
 
+// LoadEventsBySource loads the events produced by a specific handler.
+//
+// hk is the handler's identity key.
+//
+// id is the instance ID, which must be empty if the handler type does not
+// use instances.
+//
+// m is ID of a "barrier" message. If supplied, the results are limited to
+// events with higher offsets than the barrier message. If the message
+// cannot be found, UnknownMessageError is returned.
+func (r *eventStoreRepository) LoadEventsBySource(
+	ctx context.Context,
+	hk, id, d string,
+) (eventstore.Result, error) {
+	var o uint64
+
+	if d != "" {
+		var ok bool
+		o, ok = r.db.event.messageIDs[d]
+		if !ok {
+			return nil, eventstore.UnknownMessageError{
+				MessageID: d,
+			}
+		}
+
+		o = o + 1
```

Comment:
🐧 
```suggestion
		// Increment the offset to select all messages after the barrier
		// message exclusively.
		o++
```

Just to make this match the SQL impl - you had the comment explaining why there, and I thought that was good.